### PR TITLE
(@theme-ui/color-modes): migrate to typescript

### PR DESCRIPTION
**Pull request by beerose**
Part of https://github.com/system-ui/theme-ui/issues/668.

### CODE REVIEW

File: packages/css/src/types.ts [hunk 1]
```diff
@@ -641,6 +641,7 @@ export interface Theme {
    * Define the colors that are available through this theme
    */
   colors?: ColorModesScale
+  rawColors?: ColorModesScale
```
Comment (hasparus) [inline packages/css/src/types.ts:644]: It's only mentioned once in the code, and it's missing from the Theme Spec in the docs.
I'd guess we could keep this as undocumented feature or maybe even drop it? @jxnblk 

It seems to me that `rawColors` may have been left here for compatibility after https://github.com/system-ui/theme-ui/pull/561

---

https://github.com/system-ui/theme-ui/blob/a96e723bad9c0d8b03c858698045acf01587cf9d/packages/color-modes/src/custom-properties.js#L76

https://theme-ui.com/theme-spec

Comment (beerose) [reply packages/css/src/types.ts:644]: I'll remove it from `Theme` type and work around here. There you can drop it whenever you want.

Comment (jxnblk) [reply packages/css/src/types.ts:644]: Yup, I think that's only there for backwards compatibility, and I suspect it's been long enough that we could drop that in v0.4

Comment (hasparus) [reply packages/css/src/types.ts:644]:
```suggestion
```

File: packages/color-modes/src/custom-properties.ts [hunk 2]
```diff
@@ -74,7 +79,7 @@ export const createColorStyles = (theme = {}) => {
     })(theme)
   }
   const colors = theme.rawColors || theme.colors
```
Comment (hasparus) [inline packages/color-modes/src/custom-properties.ts:81]:
```suggestion
  const { colors } = theme
```

**Pull request merged**